BlazarBase/src/main/java/com/hubspot/blazar/base/BuildCommand.java

```diff
@@ -0,0 +1,61 @@
+package com.hubspot.blazar.base;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Objects;
+import com.google.common.collect.Lists;
+
+public class BuildCommand {
+  private static String DEFAULT_CMD = "/bin/bash";
+  private static String DEFAULT_ARG = "-c";
+
+  private final String executable;
+  private final List<String> args;
+  private final List<Integer> returnCodes;
+  private final Map<String, String> commandSpecificEnvironment;
+
+  @JsonCreator
+  public BuildCommand(@JsonProperty("executable") String executable,
+                      @JsonProperty("args") List<String> args,
+                      @JsonProperty("returnCodes") List<Integer> returnCodes,
+                      @JsonProperty("env") Map<String, String> commandSpecificEnvironment) {
+    this.executable = Objects.firstNonNull(executable, DEFAULT_CMD);
+    this.args = Objects.firstNonNull(args, Lists.newArrayList(DEFAULT_ARG));
```

Comment:
Why not? This way you can specify an `env` block or `returnCodes` without needing to specify your executable, which is likely to be more common than wanting to customize your executable.